src/model/CoilCoolingDXCurveFitSpeed.hpp

```diff
@@ -0,0 +1,197 @@
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_COILCOOLINGDXCURVEFITSPEED_HPP
+#define MODEL_COILCOOLINGDXCURVEFITSPEED_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ResourceObject.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class Curve;
+class CoilCoolingDXCurveFitOperatingMode;
+
+namespace detail {
+
+  class CoilCoolingDXCurveFitSpeed_Impl;
+
+} // detail
+
+/** CoilCoolingDXCurveFitSpeed is a ResourceObject that wraps the OpenStudio IDD object 'OS:Coil:Cooling:DX:CurveFit:Speed'. */
+class MODEL_API CoilCoolingDXCurveFitSpeed : public ResourceObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit CoilCoolingDXCurveFitSpeed(const Model& model,
+    Curve& totalCoolingCapacityModifierFunctionofTemperatureCurve,
+    Curve& totalCoolingCapacityModifierFunctionofAirFlowFractionCurve,
+    Curve& energyInputRatioModifierFunctionofTemperatureCurve,
+    Curve& energyInputRatioModifierFunctionofAirFlowFractionCurve,
+    Curve& partLoadFractionCorrelationCurve,
+    Curve& wasteHeatModifierFunctionofTemperatureCurve);
+
+  /** Create CoilCoolingDXCurveFitSpeed with default curves **/
+  explicit CoilCoolingDXCurveFitSpeed(const Model& model);
+
+  virtual ~CoilCoolingDXCurveFitSpeed() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  /** @name Getters */
+  //@{
+
+  double grossTotalCoolingCapacityFraction() const;
+
+  double evaporatorAirFlowRateFraction() const;
+
+  double condenserAirFlowRateFraction() const;
+
+  boost::optional<double> grossSensibleHeatRatio() const;
+
+  bool isGrossSensibleHeatRatioAutosized() const;
+
+  double grossCoolingCOP() const;
+
+  double activeFractionofCoilFaceArea() const;
+
+  double ratedEvaporatorFanPowerPerVolumeFlowRate() const;
+
+  double evaporativeCondenserPumpPowerFraction() const;
+
+  double evaporativeCondenserEffectiveness() const;
+
+  Curve totalCoolingCapacityModifierFunctionofTemperatureCurve() const;
+
+  Curve totalCoolingCapacityModifierFunctionofAirFlowFractionCurve() const;
+
+  Curve energyInputRatioModifierFunctionofTemperatureCurve() const;
+
+  Curve energyInputRatioModifierFunctionofAirFlowFractionCurve() const;
+
+  Curve partLoadFractionCorrelationCurve() const;
+
+  Curve wasteHeatModifierFunctionofTemperatureCurve() const;
```

Comment:
Hum so it seems you are actually making all of them required when the IDD isn't marked `\required-field.` For the first 5 I tend to agree. For wasteHeatModifierFunctionofTemperatureCurve it must be optional, all E+ examples have it blanked.

Reply:
Done.